src/commonMain/kotlin/com/charleskorn/kaml/YamlNode.kt

```diff
@@ -248,3 +250,22 @@ public data class YamlTaggedNode(val tag: String, val innerNode: YamlNode) : Yam
 
     override fun toString(): String = "tagged '$tag': $innerNode"
 }
+
+public val YamlNode.yamlScalar: YamlScalar
+    get() = this as? YamlScalar ?: error(this, "YamlScalar")
+
+public val YamlNode.yamlNull: YamlNull
+    get() = this as? YamlNull ?: error(this, "YamlNull")
+
+public val YamlNode.yamlList: YamlList
+    get() = this as? YamlList ?: error(this, "YamlList")
+
+public val YamlNode.yamlMap: YamlMap
+    get() = this as? YamlMap ?: error(this, "YamlMap")
+
+public val YamlNode.yamlTaggedNode: YamlTaggedNode
+    get() = this as? YamlTaggedNode ?: error(this, "YamlTaggedNode")
```

Comment:
Perhaps these would be clearer named like `asScalar`, `asNull`, `asList` etc.?

Reply:
This was also deliberate to mirror the extension properties from kotlinx.serialization.json https://github.com/Kotlin/kotlinx.serialization/blob/43d5f7841fc744b072a636b712e194081456b5ba/formats/json/commonMain/src/kotlinx/serialization/json/JsonElement.kt#L148-L174

Reply:
Makes sense to me - thanks for clarifying.